### PR TITLE
Avoid partitioning static CUDA graph kernels as dynamic (#197139)

### PR DESCRIPTION
**Pull request by Microve**
Summary:

When `cudagraph_skip_dynamic_graphs` is enabled, graph partitioning
currently treats kernel-local index symbols as evidence of dynamic shapes.
This partitions static pointwise kernels out of CUDA graphs.

for example: 

Loops._index() creates an internal SymPy index:
```
i0 = sympy_index_symbol_with_prefix(SymT.INDEX, 0)
inner_fn(i0)
```
Symbol analysis then returns:
```
{i0}
```
The old check sees that nonempty set and reports "dynamic shape ops".

Test Plan:
**Without this change**
We saw that no triton kernels included in the partition
https://www.internalfb.com/mlhub/pipelines/runs/mast/aps-f1142140950-1142154274?job_attempt=0&version=0&tab=summary&env=PRODUCTION
[tlparse](https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/aps-f1142140950-1142154274/attempt_0/version_0/rank_0/-_30_0_1/inductor_output_code_cjf5rndt4qndiqf46gnwisfjtnovu3iw2bm3jp3cow7mvvxgt4dy_510.txt?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000)

for example: [triton_poi_fused_add_cat_mul_periodically_sync_all_grads_view_72](https://www.internalfb.com/phabricator/paste/view/P2506104613) is static kernel but not included in the partition

**With this change**
https://www.internalfb.com/mlhub/pipelines/runs/mast/aps-f1144204450-1144225436?job_attempt=0&version=0&tab=summary&env=PRODUCTION
static triton kernels are included in the partition
[tlparse](https://manifold.edge.x2p.facebook.net/v0/read/tree/logs/aps-f1144204450-1144225436/attempt_0/version_0/rank_0/-_30_0_1/inductor_output_code_cbdkjo3uxtnanwgqmkssmyfbknmpmbxe33wwqxub4cuvnxh4pzh2_882.txt?bucketName=tlparse_reports&apiKey=tlparse_reports-key&withPayload=1&timeoutMsec=10000)

Differential Revision: D120192080

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

File: torch/_inductor/scheduler.py [hunk 1]
```diff
@@ -11223,7 +11242,7 @@ def should_partition(self, node: BaseSchedulerNode) -> str | None:
 
         # Partition around nodes with dynamic shapes when cudagraph_skip_dynamic_graphs is enabled
         if config.triton.cudagraph_skip_dynamic_graphs:
-            if get_scheduler_node_symbol_uses(node):
+            if filter_graph_level_symbols(get_scheduler_node_symbol_uses(node)):
```
Comment (eellison) [inline torch/_inductor/scheduler.py:11250]: > AI-assisted review (Codex):
>
> This filter drops `SymT.PRECOMPUTED_SIZE`, which can hide a dynamic dependency. For static `x` and dynamic `y`, this has a static output shape, but its indexing depends on the precomputed symbol `ps0`:
>
> ```python
> import torch
>
>
> @torch.compile(
>     fullgraph=True,
>     options={
>         "graph_partition": True,
>         "triton.cudagraphs": True,
>         "triton.cudagraph_skip_dynamic_graphs": True,
>     },
> )
> def fn(x, y):
>     p = y.shape[0] // 2
>     return torch.nn.functional.pad(x, (p, -p))
>
>
> x = torch.randn(32, device="cuda")
> y = torch.randn(8, device="cuda")
> torch._dynamo.mark_static(x)
> torch._dynamo.mark_dynamic(y, 0)
> fn(x, y)
> ```
>
> With CUDA graphs, graph partitioning, and `cudagraph_skip_dynamic_graphs=True`, dropping `ps0` admits this node into a CUDA graph partition. The generated partition computes `ps0 = s17 // 2` without receiving `s17`, raising `NameError`. The reproduction passes with the base scheduler methods and fails with the PR methods on the local build.
>
> Could we expand precomputed aliases using `sizevars.remove_precomputed_replacements` in `get_scheduler_node_symbol_uses`, then collect their free symbols? That gives both dynamic-shape classification and partition argument generation the underlying dependency. Please add the padding example as regression coverage.

**Review by eellison — COMMENTED** [review]
looks good, one comment

**Review by eellison — APPROVED** [review]

**Comment by nWEIdia** [comment]
The B200 failure after landing is due to infra/broken B200 accidentally brought online, sorry, please ignore that signal. I will retry it later. https://github.com/pytorch/pytorch/actions/runs/35384040673/job/105730556711
